ffi/src/logging.rs

```diff
@@ -78,29 +79,27 @@ impl LogArgs<'_> {
             })?;
         }
 
-        let level = self.log_level()?;
-        let level = level.parse().map_err(|e| {
-            std::io::Error::new(
-                std::io::ErrorKind::InvalidInput,
-                format!("invalid log level `{level}`: {e}"),
-            )
-        })?;
+        let filter = self.log_filter()?;
 
         let file = OpenOptions::new()
             .create(true)
-            .write(true)
-            .truncate(false)
-            .open(&log_path)
+            .append(true)
+            .open(log_path)
             .map_err(|e| {
                 std::io::Error::new(
                     e.kind(),
                     format!("failed to open log file `{}`: {e}", log_path.display()),
                 )
             })?;
 
-        env_logger::Builder::new()
-            .filter_level(level)
-            .target(Pipe(Box::new(file)))
+        let mut builder = env_logger::Builder::new();
+        builder.target(Pipe(Box::new(file)));
```

Comment:
We should wrap `file` with `std::io::BufWriter` to make using `stdout` slightly more copacetic (doesn't prevent interleaving but makes it harder if we only write whole lines). `env_logger` calls `flush` after writing out every event.